Orange/classification/_tree_scorers.pyx

```diff
@@ -0,0 +1,403 @@
+
+    Args:
+        x: attribute values
+        y: target values
+        n_values: the number of attribute values
+        min_leaf: the minimal number of instances on each side of the threshold
+
+    Returns:
+        MSE decrease
+    """
+
+    cdef:
+        int i, n
+        double sum = 0, inter, tx
+
+        np.int32_t[:] group_sizes = numpy.zeros(n_values, dtype=numpy.int32)
+        double[:] group_sums = numpy.zeros(n_values)
+
+    with nogil:
+        for i in range(x.shape[0]):
+            tx = x[i]
+            if not npy_isnan(tx):
+                group_sizes[<int>tx] += 1
+                group_sums[<int>tx] += y[i]
+        inter = 0
+        n = 0
+        for i in range(n_values):
+            if group_sizes[i] < min_leaf:
+                # We don't construct nodes with less than min_leaf instances
+                # If there is only one non-null node, the split will yield a
+                # score of 0
+                continue
+            inter += group_sums[i] * group_sums[i] / group_sizes[i]
+            sum += group_sums[i]
+            n += group_sizes[i]
+    if n < 2:
+        return 0
+    # factor n / x.shape[0] is the punishment for missing values
+    #return (inter - sum * sum / n) / n * n / x.shape[0]
+    return (inter - sum * sum / n) / x.shape[0]
+
+
+def compute_predictions(double[:, :] X, int[:] code,
+                        double[:, :] values, double[:] thresholds):
+    """
+    Return the values (distributions, means and variances) stored in the nodes
+    to which the tree classify the rows in X.
+
+    The tree is encoded by :obj:`Orange.tree.OrangeTreeMode._compile`.
+
+    The result is a matrix of shape (X.shape[0], values.shape[1])
+
+    Args:
+        X: data for which the predictions are made
+        code: encoded tree
+        values: values corresponding to tree nodes
+        thresholds: thresholds for numeric nodes
+
+    Returns:
+        a matrix of values
+    """
+    cdef:
+        unsigned int node_ptr, i, val_idx
+        signed int next_node_ptr, node_idx
+        np.float64_t val
+        double[: ,:] predictions = numpy.empty(
+            (X.shape[0], values.shape[1]), dtype=numpy.float64)
+        # np.ndarray[np.float64_t, ndim=2] predictions = numpy.empty(
+        #     (X.shape[0], class_distrs.shape[1]), dtype=numpy.float64)
+
+    for i in range(X.shape[0]):
+        node_ptr = 0
+        while code[node_ptr]:
+            val = X[i, code[node_ptr + 2]]
+            if npy_isnan(val):
+                break
+            if code[node_ptr] == 3:
+                node_idx = code[node_ptr + 1]
+                val_idx = int(val > thresholds[node_idx])
+            else:
+                val_idx = int(val)
+            next_node_ptr = code[node_ptr + 3 + val_idx]
+            if next_node_ptr == NULL_BRANCH:
+                break
+            node_ptr = next_node_ptr
+        node_idx = code[node_ptr + 1]
+        predictions[i] = values[node_idx]
```

Comment:
```cython
for j in range(values.shape[1]):
    predictions[i, j] = values[node_idx, j]
```
Will allow it to be `nogil` and is just what `__pyx_memoryview_copy_contents()` does otherwise.